third_party/move/move-model/bytecode/src/astifier.rs

```diff
@@ -548,39 +533,84 @@ impl Generator {
-            let mut group = top_sort.pop_all();
+            // Peek the group of blocks which are available according to the partial order.
+            // We select one of them below.
+            let group = top_sort.peek_all();
             if group.is_empty() {
                 break;
             }
-            // Sort the members of the group (1) by priority of the block terminator. We want
-            // to have jumps and other exits first (2) by code offset as it was determined
-            // by the DFS ordering of blocks in cleanup.
-            let terminator_priority = |blk: BlockId| -> isize {
+            // Select the smallest of the group according to the following order:
+            // (1) A block belonging to an active loop is smaller.
+            // (2) Next, according the way blocks are terminated. This is semantically
+            //     not relevant, but leads to better readable code, e.g. abort
+            //     first.
+            // (2) Finally, which is earlier in the code as produced by the DFS ordering of blocks
+            //     in cleanup.
+            let current_loop = blocks
+                .last()
+                .and_then(|blk_id| ctx.label_of_block(*blk_id))
+                .and_then(|l| ctx.block_to_header.get(&l))
+                .cloned();
+            let loop_priority = |b1: BlockId| -> usize {
+                if current_loop
+                    .and_then(|header| {
+                        ctx.label_of_block(b1)
+                            .map(|l| ctx.loop_labels[&header].contains(&l))
+                    })
+                    .unwrap_or(false)
+                {
+                    // Block has higher priority since it belongs to active loop
+                    0
+                } else {
+                    1
+                }
+            };
+            let terminator_priority = |blk: BlockId| -> usize {
                 use Bytecode::*;
                 match ctx.code_for_block(blk).last() {
                     None => 0,
-                    Some(Jump(..)) => 1,
-                    Some(Abort(..)) => 2,
-                    Some(Ret(..)) => 3,
-                    Some(Branch(..)) => 4,
+                    Some(Abort(..)) => 1,
+                    Some(Jump(..)) => 2,
+                    Some(Branch(..)) => 3,
+                    Some(Ret(..)) => 4,
                     _ => panic!("unexpected block terminator"),
                 }
             };
-            group.sort_by(|b1, b2| {
-                let c = terminator_priority(*b1).cmp(&terminator_priority(*b2));
-                if c.is_eq() {
-                    // Use code offset
-                    ctx.forward_cfg
-                        .code_range(*b1)
-                        .start
-                        .cmp(&ctx.forward_cfg.code_range(*b2).start)
-                } else {
-                    c
+            let min = group
+                .iter()
+                .copied()
+                .copied()
```

Comment:
Remove double `.copied()` here.

Reply:
It's actually needed, because the underlying thing is `&&T`.